src/categorical_algebra/CSets.jl

```diff
@@ -453,8 +453,12 @@ end
   
 function coerce_component(ob::Symbol, f::FinFunction{Int,Int},
                           dom_size::Int, codom_size::Int; kw...)
-  length(dom(f)) == dom_size || error("Domain error in component $ob")
-  # length(codom(f)) == codom_size || error("Codomain error in component $ob") # codom size is now Maxpart not nparts
+  if haskey(kw, :dom_parts)
+    !any(i -> f(i) == 0, kw[:dom_parts]) # check domain of mark as deleted
+  else                         
+    length(dom(f)) == dom_size # check domain of dense parts
+  end || error("Domain error in component $ob")
+  # length(codom(f)) == codom_size || error("Codomain error in component $ob")
```

Comment:
Looks like this code has been dead for some time now. Can we just delete it?

Reply:
Yeah, I'll add a comment that (for whatever reason) we check domains but not codomains in `coerce_component`